lib/internal/http2/core.js

```diff
@@ -0,0 +1,2392 @@
+    if (this[kState].destroyed)
+      throw new errors.Error('ERR_HTTP2_INVALID_SESSION');
+
+    if (!(stream instanceof Http2Stream)) {
+      throw new errors.TypeError('ERR_INVALID_ARG_TYPE',
+                                 'stream',
+                                 'Http2Stream');
+    }
+    assertIsObject(options, 'options');
+    options = Object.assign(Object.create(null), options);
+    validatePriorityOptions(options);
+
+    debug(`[${sessionName(this[kType])}] sending priority for stream ` +
+          `${stream[kID]}`);
+
+    // A stream cannot be made to depend on itself
+    if (options.parent === stream[kID]) {
+      debug(`[${sessionName(this[kType])}] session still connecting. queue ` +
+            'priority');
+      throw new errors.TypeError('ERR_INVALID_OPT_VALUE',
+                                 'parent',
+                                 options.parent);
+    }
+
+    if (stream[kID] === undefined) {
+      stream.once('ready', submitPriority.bind(this, stream, options));
+      return;
+    }
+    submitPriority.call(this, stream, options);
+  }
+
+  // Submits an RST-STREAM frame to be sent to the remote peer. This will
+  // cause the stream to be closed.
+  rstStream(stream, code = NGHTTP2_NO_ERROR) {
+    if (this[kState].destroyed)
+      throw new errors.Error('ERR_HTTP2_INVALID_SESSION');
+
+    if (!(stream instanceof Http2Stream)) {
+      throw new errors.TypeError('ERR_INVALID_ARG_TYPE',
+                                 'stream',
+                                 'Http2Stream');
+    }
+
+    if (typeof code !== 'number') {
+      throw new errors.TypeError('ERR_INVALID_ARG_TYPE',
+                                 'code',
+                                 'number');
+    }
+
+    if (this[kState].rst) {
+      // rst has already been called, do not call again,
+      // skip straight to destroy
+      stream.destroy();
+      return;
+    }
+    stream[kState].rst = true;
+    stream[kState].rstCode = code;
+
+    debug(`[${sessionName(this[kType])}] initiating rststream for stream ` +
+          `${stream[kID]}: ${code}`);
+
+    if (stream[kID] === undefined) {
+      debug(`[${sessionName(this[kType])}] session still connecting, queue ` +
+            'rststream');
+      stream.once('ready', submitRstStream.bind(this, stream, code));
+      return;
+    }
+    submitRstStream.call(this, stream, code);
+  }
+
+  // Destroy the Http2Session
+  destroy() {
+    const state = this[kState];
+    if (state.destroyed || state.destroying)
+      return;
+
+    debug(`[${sessionName(this[kType])}] destroying nghttp2session`);
+    state.destroying = true;
+
+    // Unenroll the timer
+    unenroll(this);
+
+    // Shut down any still open streams
+    const streams = state.streams;
+    streams.forEach((stream) => stream.destroy());
+
+    // Disassociate from the socket and server
+    const socket = this[kSocket];
+    // socket.pause();
+    delete this[kSocket];
+    delete this[kServer];
```

Comment:
ugh. checking out those updates was further down in my todo list today. That is really unfortunate :-/